packages/vite/src/client/client.ts

```diff
@@ -15,7 +15,7 @@ console.log('[vite] connecting...')
 // use server configuration, then fallback to inference
 const socketProtocol =
   __HMR_PROTOCOL__ || (location.protocol === 'https:' ? 'wss' : 'ws')
-const socketHost = `${__HMR_HOSTNAME__ || location.hostname}:${__HMR_PORT__}`
+const socketHost = `${__HMR_HOSTNAME__ || location.hostname}:${location.port}`
```

Comment:
```suggestion
const socketHost = `${__HMR_HOSTNAME__ || location.hostname}:${__HMR_PORT__ || location.port}`
```

Think this could be better. Respect `server.hmr.port` if specified explicitly or use the same port as client. `clientInject.ts` file needs to be changed accordingly.